presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkSessionProperties.java

```diff
@@ -138,6 +139,15 @@ public PrestoSparkSessionProperties(PrestoSparkConfig prestoSparkConfig)
                         "When revoking memory, try to revoke so much that memory pool is filled below target at the end",
                         prestoSparkConfig.getMemoryRevokingTarget(),
                         false),
+                new PropertyMetadata<>(
+                        SPARK_EXECUTION_STRATEGIES,
+                        "Execution strategies to be applied while running the query",
+                        VARCHAR,
+                        List.class,
+                        prestoSparkConfig.getExecutionStrategies(),
+                        false,
+                        value -> Splitter.on(',').trimResults().omitEmptyStrings().splitToList(value.toString()),
```

Comment:
Does it make sense to validate the passed strategies to make sure that these are valid? Currently, we do this validation later and basically skips the invalid values. We should basically fail if the passed strategies are incorrect immediately.  Maybe we can extend the decoder function and perform validation inside decoder itself?
```
    private List<ExecutionStrategy> getExecutionStrategies(Map<String, String> sessionProperties)
    {
        Set<String> allExecutionStrategies = Arrays.stream(ExecutionStrategy.values()).map(v -> v.name()).collect(Collectors.toSet());
        String executionStrategies = sessionProperties.getOrDefault(SPARK_EXECUTION_STRATEGIES, "");
        return Splitter.on(',').trimResults().omitEmptyStrings().splitToList(executionStrategies)
                .stream()
                .filter(tag -> allExecutionStrategies.contains(tag))
                .map(t -> ExecutionStrategy.valueOf(t))
                .collect(Collectors.toList());
    }
```

Reply:
Yes, added validator